### PR TITLE
Csp 220 hide course management links and text unless on beta list

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Services/GetRoatpBetaProviderService.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        public bool IsUkprnEnabled(int ukprn)
+        {
+            var featureToggles = _roatpCourseManagementWebConfiguration.ProviderFeaturesConfiguration.FeatureToggles;
+
+            var courseManagementFeature = featureToggles.FirstOrDefault(f => f.Feature == CourseManagement);
+            if (courseManagementFeature == null) return false;
+
+            var providerUkrpns= !courseManagementFeature.IsEnabled 
+                                        || courseManagementFeature?.Whitelist == null ?
```
Comment (DevenKShah) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Services/GetRoatpBetaProviderService.cs:26]: The formatting here is weird

File: src/SFA.DAS.ProviderApprenticeshipsService.Web.UnitTests/Services/GetRoatpBetaProviderServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        public void IsUkprnEnabled_DifferentFeatureName_ReturnsFalse()
+        {
+            _configuration.ProviderFeaturesConfiguration.FeatureToggles[0].Feature = "DifferentName";
+            _service = new GetRoatpBetaProviderService(_configuration);
+            var result = _service.IsUkprnEnabled(ukprnPresent);
+            Assert.IsFalse(result);
+        }
+
+        [Test]
+        public void IsUkprnEnabled_NotIsEnabled_ReturnsFalse()
```
Comment (DevenKShah) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web.UnitTests/Services/GetRoatpBetaProviderServiceTests.cs:54]: IsUkprnEnabled_FeatureNotEnabled_ReturnsFalse

**Review by DevenKShah — APPROVED** [review]
I have added some minor comments and few questions, I need to understand the register class better I guess.

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/DependencyResolution/RoatpCourseManagementWebRegistry.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        private string GetAndStoreEnvironment()
+        {
+            var environment = Environment.GetEnvironmentVariable("DASENV");
+            if (string.IsNullOrEmpty(environment))
+            {
+                environment = ConfigurationManager.AppSettings["EnvironmentName"];
+            }
+            if (environment.Equals("LOCAL") || environment.Equals("AT") || environment.Equals("TEST"))
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/DependencyResolution/RoatpCourseManagementWebRegistry.cs:34]: yes

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/DependencyResolution/RoatpCourseManagementWebRegistry.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            {
+                PopulateSystemDetails(environment);
+            }
+        
+            return environment;
+        }
+
+        private RoatpCourseManagementWebConfiguration GetConfiguration(string environment, IConfigurationRepository configurationRepository)
+        {
+            var configurationService = new ConfigurationService(configurationRepository,
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/DependencyResolution/RoatpCourseManagementWebRegistry.cs:44]: yes

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Services/GetRoatpBetaProviderService.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Services/GetRoatpBetaProviderService.cs:26]: resolved

**Review by ChrisJWoodcock — COMMENTED** [review]
The service storage configuration file of a different service is being accessed from PAS in order to gain access to its feature flags and whitelisting to avoid config duplication but accessing the config of a different service creates a dependency between the two services. This has been explained as temporary code which will be removed later but we have lots of that which hasn't been removed. Others may build on this - note the inclusion of the database connection string to the other service - imagine someone using that!

**Comment by DevenKShah** [comment]
> note the inclusion of the database connection string to the other service - imagine someone using that!
The connection string in our web config is for shared redis cache. We have taken the comments on board and we will be removing this code in near future with the planned work.

**Pull request merged**